aiida/orm/nodes/data/remote/stash.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+"""Data plugin that models an archived folder on a remote computer."""
+import typing
+
+from aiida.common.datastructures import StashMode
+from aiida.common.lang import type_check
+from ..data import Data
+
+__all__ = ('RemoteStashData',)
+
+
+class RemoteStashData(Data):
+    """Data plugin that models an archived folder on a remote computer.
+
+    A stashed folder is essentially an instance of ``RemoteData`` that has been archived. Archiving in this context can
+    simply mean copying the content of the folder to another location on the same or another filesystem as long as it is
+    on the same machine. In addition, the folder may have been compressed into a single file for efficiency or even
+    written to tape. The ``stash_mode`` attribute will distinguish how the folder was stashed which will allow the
+    implementation to also `unstash` it and transform it back into a ``RemoteData`` such that it can be used as an input
+    for new ``CalcJobs``.
+    """
+
+    def __init__(
+        self, stash_mode: StashMode = None, target_basepath: str = None, stash_list: typing.List = None, **kwargs
```

Comment:
Caught another bug during field testing: 🙃

```suggestion
        self, stash_mode: StashMode = None, target_basepath: str = None, source_list: typing.List = None, **kwargs
```

Reply:
looks like I have a test to add :)